resource-managers/mesos/src/main/scala/org/apache/spark/scheduler/cluster/mesos/MesosCoarseGrainedSchedulerBackend.scala

```diff
@@ -213,6 +216,24 @@ private[spark] class MesosCoarseGrainedSchedulerBackend(
       sc.conf.getOption("spark.mesos.driver.frameworkId").map(_ + suffix)
     )
 
+    // check that the credentials are defined, even though it's likely that auth would have failed
+    // already if you've made it this far
+    if (principal != null && currentHadoopDelegationTokens.isDefined) {
+      logDebug(s"Principal found ($principal) starting token renewer")
+      // The renewal time is ignored when getting the initial delegation tokens
+      // (CoarseGrainedSchedulerBackend.scala:getHadoopDelegationCreds), so we get the renewal
+      // time here and schedule a thread to renew them.
+      val renewalTime =
```

Comment:
I still don't like this. You should not need to implement this separate method of getting the renewal time just because the current code is throwing out that information. Instead you should fix the code so that the information is preserved.

`getHadoopDelegationCreds` is called in only one place, so my suggestion would be to encapsulate initializing the token manager and getting the initial set of tokens into a single method (instead of the current two).

Then in that method's implementation you can get the initial set of tokens, initialize the renewer thread with the correct renewal period, and return the data needed by the scheduler.

Reply:
Ok, I agree. I refrained from touching the current code only because it was added it so recently. I assumed the omission of renewal time was intentional (or at least semi-intentional). I'll submit a refactor. Are you suggesting that we make the renewal thread part of `CoarseGrainedSchedulerBackend` (with the appropriate override for YARN/Mesos)?  Or make a resource-manager specific method?

Reply:
It makes more sense to have it in `CoarseGrainedSchedulerBackend` (then YARN and Mesos can easily share it); my worry is exposing the functionality to standalone (it would work, but not be secure, which is a problem).

So maybe add a second method `def supportsTokenRenewal` that subclasses can override to return `true` or something.

Reply:
Maybe an overly simplistic solution, but what about putting this functionality into `HadoopDelegationTokenManager`?

Reply:
I guess that would be ok. I was a little worried about mixing the two but they are kinda closely related...

Reply:
But do note that you still can't start the token renewer automatically, since YARN still has its own renewer and things would get mixed up.